### ISSUE TITLE
Deploying web contents with large numbers of files -- some rate control needed

### ISSUE THREAD

**Issue by joshuaauerbachwatson** [nimbella/nimbella-cli#35]
Transient errors (socket hangups etc) are sometimes observed when deploying the web content of projects that have a "large number" (100s) of files to upload.   I attribute this to the Google cloud storage server having a limit on now many open requests it can handle.   

A likely solution is to cap the number of web upload Promises that are outstanding at a time.  Currently, the deployer accumulates the array of Promises for all the web resources and settles them as part of a single `Promise.all`.    This makes it vulnerable to the size of the web resource set.

**Comment by joshuaauerbachwatson** [comment]
I have observed frequent failures (socket hangups) deploying the election demo (it has 100+ web files).  It doesn't fail every time but often enough to be noticed.   I was pretty sure it was due to this issue so I upped the priority and got this fixed.

The fix is pretty simplistic: I introduced a web "chunk size" constant and deploy web content in chunks of that size (currently 20).   So, at most 20 web upload requests are outstanding.   Throughput may be suboptimal, because the deployer waits for all 20 promises resulting from chunk 1 to settle before starting chunk 2.   But, it seems to work.  All regression tests pass and the election demo deploy stopped failing.   My impression (very very anecdotal) is that throughput when deploying this largest of our demos does not really suffer compared to the old way.   I think the old way may have caused some backoff/retry that actually slowed things down.  Not to mention that it often failed.

There are some packages out there (e.g. es6-promise-pool) that claim to do this kind of thing while keeping a specified number of promises outstanding (so, e.g., we would really be fixing the concurrency level at, say, 20).   It's a much bigger hit to the code to reorganize around using that.   I can imagine some future effort to address concurrency throughout the deployer where we might adopt that technology.   For now, I think this is good enough.

This now awaits a stable version.
